### PR TITLE
i183: Reduce boiler-plate code for AddProblem

### LINKED ISSUE #183: Reduce boiler-plate code for AddProblem

### PR DESCRIPTION
**Pull request by T-Davis**
Fixes #183

### CODE REVIEW

**Review by tamari-gray — APPROVED** [review]
insane trev 🤯 
beautiful work! Just that comment, about possibly reusing a function. Not blocking a sweet approve on a veddy sweet pr. thanks!

File: lib/services/auth_service.dart [hunk 1]
```diff
@@ -117,12 +110,11 @@ class AuthService {
...
         default:
-          yield AddProblem(
-            problem: AppleSignInProblem.by(
-              (b) => b
-                ..message = e.toString()
-                ..info = BuiltMap<String, String>({'code': e.code}),
-            ),
+          yield createAddProblem(
+            AppleSignInProblem,
+            e,
```
Comment (nickmeinhold) [inline lib/services/auth_service.dart:115]: Do you think we should change `e` to `error`?  I *think* we tend to use `error` and it'd be good to be consistent.  I'm aware the old code uses `e` but given we're making a change anyway, what do you think?

File: lib/services/auth_service.dart [hunk 2]
```diff
@@ -166,24 +152,13 @@ class AuthService {
...
     } on FirebaseAuthException catch (e) {
       yield UpdateEmailAuthOptionsPage(step: AuthStep.waitingForInput);
-      yield AddProblem(
-        problem: EmailSignInProblem.by(
-          (b) => b
-            ..message = e.message.toString()
-            ..info = BuiltMap<String, String>({'code': e.code}),
-        ),
-      );
+      yield createAddProblem(EmailSignInProblem, e.message, StackTrace.current,
```
Comment (nickmeinhold) [inline lib/services/auth_service.dart:155]: e -> error, if you agree

File: lib/services/auth_service.dart [hunk 3]
```diff
@@ -211,24 +186,13 @@ class AuthService {
...
     } on FirebaseAuthException catch (e) {
       yield UpdateEmailAuthOptionsPage(step: AuthStep.waitingForInput);
-      yield AddProblem(
-        problem: EmailSignUpProblem.by(
-          (b) => b
-            ..message = e.message.toString()
-            ..info = BuiltMap<String, String>({'code': e.code}),
-        ),
-      );
+      yield createAddProblem(EmailSignUpProblem, e.message, StackTrace.current,
```
Comment (nickmeinhold) [inline lib/services/auth_service.dart:189]: e -> error, if you agree

**Review by nickmeinhold — APPROVED** [review]
Amazing effort Trev!!!!  Man that was beautiful seeing all those big red blocks with a single green line below 😄

**Pull request merged**